gstreamer-pbutils/src/auto/discoverer.rs

```diff
@@ -0,0 +1,149 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 47eb915)
+// from gir-files (https://github.com/gtk-rs/gir-files @ ???)
+// DO NOT EDIT
+
+use DiscovererInfo;
+use Error;
+use ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use gst_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct Discoverer(Object<ffi::GstDiscoverer, ffi::GstDiscovererClass>);
+
+    match fn {
+        get_type => || ffi::gst_discoverer_get_type(),
+    }
+}
+
+impl Discoverer {
+    pub fn new(timeout: gst::ClockTime) -> Result<Discoverer, Error> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::gst_discoverer_new(timeout.to_glib(), &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+}
+
+unsafe impl Send for Discoverer {}
+unsafe impl Sync for Discoverer {}
+
+pub trait DiscovererExt {
+    fn discover_uri(&self, uri: &str) -> Result<DiscovererInfo, Error>;
+
+    fn discover_uri_async(&self, uri: &str) -> Result<(), glib::error::BoolError>;
```

Comment:
This is never going to return an Error but I'd prefer to keep it as is to be in sync with the C API.

Reply:
Ok